climada/hazard/tc_tracks.py

```diff
@@ -799,6 +805,56 @@ def read_netcdf(self, folder_name):
             track.attrs['orig_event_flag'] = bool(track.orig_event_flag)
             self.data.append(track)
 
+    def to_geodataframe(self, as_points=False, joint=True):
+        """Transform this TCTracks instance into a GeoDataFrame.
+
+        Parameters:
+            as_points (bool): If true, construct LineString or single Point
+                geometries. A point is created if a track has length one.
+                If set to false, the geometries are returned as points with an
+                additional timestamp column.
+            joint (bool): If using as_points, return geometries joined to
+                metadata. If set to False, return a tuple of two GeoDataFrames
+                that can be joined on their index.
```

Comment:
The meaning of parameters should ideally be independent from each other. This is not the case here. The meaning of `joint` depends on the value of `as_points`: if the latter is `True` it means something - otherwise not.

Reply:
I know that this is not optimal, but this again relates to the efficiency concerns above. I'm testing the entire ibtracs archive now to see how this performs at scale, and if the splitting afterward would indeed be as terrible as I think it is.

Reply:
Surprising! Do you have numbers?

Reply:
See comment below :)